.github/workflows/windows-ci.yml

```diff
@@ -0,0 +1,126 @@
+name: Windows-CI
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: windows-latest
+
+    strategy:
+      matrix:
+        arch: ["x86", "x64"]
+        qt: ["5.15.0"]
+
+    env:
+      CMAKE_GENERATOR: Visual Studio 16 2019
+      vcpkg-triplet: ${{ matrix.arch }}-windows-static-md
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Bootstrap vcpkg
+        id: bootstrap-vcpkg
+        run: |
+          cd C:\vcpkg
+          git pull
+          Write-Host "::set-output name=vcpkg-rev::$(git rev-parse HEAD)"
+          ./bootstrap-vcpkg.bat
+
+      - name: Restore Dependency Cache
+        id: cache-vcpkg
+        uses: actions/cache@v1
+        with:
+          path: C:\vcpkg\installed
+          key: |
+            vcpkg-triplet=${{ env.vcpkg-triplet }} vcpkg-response=${{ hashFiles('vcpkg.txt') }} vcpkg-rev=${{ steps.bootstrap-vcpkg.outputs.vcpkg-rev }}
+          restore-keys: |
+            vcpkg-triplet=${{ env.vcpkg-triplet }} vcpkg-response=${{ hashFiles('vcpkg.txt') }}
+            vcpkg-triplet=${{ env.vcpkg-triplet }}
+
+      - name: Upgrade Dependencies
+        if: steps.cache-vcpkg.outputs.cache-hit == 'true'
+        run: |
+          vcpkg upgrade --no-dry-run --triplet ${{ env.vcpkg-triplet }}
+
+      - name: Build Dependencies
+        run: |
+          vcpkg install `@vcpkg.txt --triplet ${{ env.vcpkg-triplet }}
+          vcpkg list --x-full-desc
+
+      - name: Configure Platform Arch
+        run: |
+          # Qt started providing both x86 and x64 builds for only MSVC 2019 as of 5.15. This was
+          # true only for MSVC 2017 before then.
+          [int[]]$qt_ver = "${{ matrix.qt }}".split('.')
+          if (($qt_ver[0] -eq 5) -and ($qt_ver[1] -lt 15)) {
+            $qt_compiler = "msvc2017"
+          } else {
+            $qt_compiler = "msvc2019"
+          }
+
+          if ("${{ matrix.arch }}" -eq "x86") {
+            $cmake_arch = "Win32"
+            $qt_arch = "win32_$($qt_compiler)"
+          } else {
+            $cmake_arch = "${{ matrix.arch }}"
+            $qt_arch = "win64_$($qt_compiler)_64"
+          }
+          Write-Host "::set-env name=CMAKE_GENERATOR_PLATFORM::$cmake_arch"
+          Write-Host "::set-env name=QT_ARCH::$qt_arch"
```

Comment:
> Error: Unable to process command '::set-env name=CMAKE_GENERATOR_PLATFORM::x64' successfully.
> Error: The `set-env` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
> Error: Unable to process command '::set-env name=QT_ARCH::win64_msvc2019_64' successfully.
> Error: The `set-env` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

Reply:
Yes, let's post critical deprecation notes to our blog 🙄. SIgh.

Reply:
This is fixed, but now there's some problem with upstream HSPlasmaNet.